api/api_full.go

```diff
@@ -409,8 +408,19 @@ type FullNode interface {
 	StateMinerRecoveries(context.Context, address.Address, types.TipSetKey) (bitfield.BitField, error) //perm:read
 	// StateMinerPreCommitDepositForPower returns the precommit deposit for the specified miner's sector
 	StateMinerPreCommitDepositForPower(context.Context, address.Address, miner.SectorPreCommitInfo, types.TipSetKey) (types.BigInt, error) //perm:read
-	// StateMinerInitialPledgeCollateral returns the initial pledge collateral for the specified miner's sector
+	// StateMinerInitialPledgeCollateral attempts to calculate the initial pledge collateral based on a SectorPreCommitInfo.
+	// This method uses the DealIDs field in SectorPreCommitInfo to determine the amount of verified
+	// deal space in the sector in order to perform a QAP calculation. Since network version 22 and
+	// the introduction of DDO, the DealIDs field can no longer be used to reliably determine verified
+	// deal space; therefore, this method is deprecated. Use StateMinerInitialPledgeForSector instead
+	// and pass in the verified deal space directly.
+	//
+	// Deprecated: Use StateMinerInitialPledgeForSector instead.
 	StateMinerInitialPledgeCollateral(context.Context, address.Address, miner.SectorPreCommitInfo, types.TipSetKey) (types.BigInt, error) //perm:read
```

Comment:
@rjan90 what do you think? I don't think it's unreasonable to just completely remove it from a 1.30.0, but we shouldn't do it in a 1.29.x. If the target is 1.29 then let's mark it as deprecated now and remove it in a major version bump.

Reply:
This will target [v1.30.0](https://github.com/filecoin-project/lotus/issues/12480), which will be the mandatory release for nv24, unless there is users blocked by it and it warrants a backport to the v1.29.x train.

We should mark it as deprecated in the v1.30.0 release notes and announce that it will be completely removed in the v[1.31.0 release](https://github.com/filecoin-project/lotus/issues/12344). The v1.31.0 release will be shipped alongside v1.30.0 for the nv24 upgrade but includes higher-risk items that we don't want to obligate users to upgrade to during a network upgrade.

Reply:
I've put a note about this in the changelog